pkg/apis/logging/v1/clusterlogging_types.go

```diff
@@ -38,6 +38,11 @@ type ClusterLoggingSpec struct {
 	//
 	// +nullable
 	Curation *CurationSpec `json:"curation,omitempty"`
+
+	// Specification for output tuning parameters
+	//
+	// +nullable
+	Output *OutputConfSpec `json:"output,omitempty"`
```

Comment:
i feel this should be nested under the fluentd spec instead, that's what its there for

as we expand other collectors we would nest specific output specs under those as well -- also we may not need to do that for other collectors

to me, this is awkward -- we are introducing another field that is specific to our collectors when we already have a field for this, and we also have a field under that specific for fluentd. to me this is redundant

Reply:
cc @alanconway

Reply:
@ewolinetz: This has been a difficult point all along, we need to hammer it out till we're all happy.

Here's a summary of the debate:

My view: Fluent is **not just a collector**. It is a collector *and a forwarder*. We are trying to escape from the CLO==fluentd mindset, and present a logical API that can control other (possibly distinct) collector/forwarder impls in future. We need to be clear (to ourselves and users) about the difference between _collection_ (which logs get scraped/received from the node) and _forwarding_ (where/how logs get sent off the node)

These tuning parameters affect fluentd *as a forwarder* - they affect fluent "output" config sections, which are created by the ClusterLogForwarder from `output` specs. They have no effect on which/how logs are collected, only on how they are sent to remote targets.

IMO this logically belongs on the ClusterLogForwarder API (my original proposal) which would also allow us to specify different tuning per output. I yielded to @christianvogt and @jcantrill argument (as I understand it): implementing per-output on the forwarder is adding new functionality, we don't want to sink so much effort. We already have global config via env-vars, so global config via API is a low-risk way to do the same thing in managed mode.

The Compromise: OK but we need a new section. This is not "collector" config, it can't be in a section called "collector" and described as "Specification of the Collection component for the cluster". We considered `forwarder` and `tuning` before settling on `output`.

@ewolinetz thoughts? We should keep at this till we're all happy with the result.

Reply:
So I think there are 3 options on the table:
- use existing collector.fluentd section
- use new sectoin output.fluentd (naming? forwarder, tuning)
- add clusterlogforwarder.output.tuning.fluentd section (output.fluentd is already used to define a fluentd *target*)

Reply:
We concluded to use `spec.forwarder.fluentd` for grouping the parameters.